client/upe-onboarding-wizard/upe-preview-methods-selector/__tests__/add-payment-methods-task.test.js

```diff
@@ -1,70 +1,147 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import WizardTaskContext from '../../wizard/task/context';
 import AddPaymentMethodsTask from '../add-payment-methods-task';
+import {
+	useGetAvailablePaymentMethodIds,
+	useEnabledPaymentMethodIds,
+	useSettings,
+} from '../../../data';
+import WCPaySettingsContext from '../../../settings/wcpay-settings-context';
+
+jest.mock( 'wcstripe/data', () => ( {
+	useGetAvailablePaymentMethodIds: jest.fn(),
+	useEnabledPaymentMethodIds: jest.fn(),
+	useSettings: jest.fn(),
+	useCurrencies: jest.fn(),
+} ) );
+
+const SettingsContextProvider = ( { children } ) => (
+	<WCPaySettingsContext.Provider
+		value={ { featureFlags: { multiCurrency: true }, accountFees: {} } }
+	>
+		{ children }
+	</WCPaySettingsContext.Provider>
+);
 
 describe( 'AddPaymentMethodsTask', () => {
-	it( 'should proceed to step 3 by clicking "Add payment methods" button', async () => {
-		const setCompletedMock = jest.fn();
+	beforeEach( () => {
+		useGetAvailablePaymentMethodIds.mockReturnValue( [
+			'card',
+			'bancontact',
+			'giropay',
+			'p24',
+			'ideal',
+			'sepa_debit',
+			'sofort',
+		] );
+		useSettings.mockReturnValue( {
+			saveSettings: () => Promise.resolve( true ),
+			isSaving: false,
+			isLoading: false,
+		} );
+		useEnabledPaymentMethodIds.mockReturnValue( [
+			[ 'card' ],
+			() => null,
+		] );
+	} );
```

Comment:
As a good practice we can clear the mocks after each test.
```suggestion
	} );
	
	afterEach( () => {
		jest.clearAllMocks();
	} );
```

Reply:
Also in this case the mocked functions are all cleared in the `beforeEach` 🤔